### PR TITLE
Set up GitHub Action to publish l2geth binaries

### PR DESCRIPTION
**Pull request by rrzhang139**
github actions workflow that builds l2geth and creates/uploads a release binary.

### CODE REVIEW

**Comment by ChuhanJin** [comment]
Can one of the admins verify this patch?

File: .github/workflows/releases.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  release:
+    strategy:
+      matrix:
+        goos: [linux, windows, darwin]
```
Comment (Thegaram) [inline .github/workflows/releases.yaml]: Based on the discussion [here](https://github.com/ethereum/go-ethereum/issues/26170#issuecomment-1318467453) I'm not sure windows will work. We can try and see.

File: .github/workflows/releases.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      run: sudo apt-get install zip
+
+    
+    - name: Build geth binary and create archives
+      env:
+        GOOS: ${{ matrix.goos }}
+        GOARCH: ${{ matrix.goarch }}
+      run: |
+        env GO111MODULE=on go run build/ci.go install ./cmd/geth
+        binary_name="geth-${GOOS}-${GOARCH}${{ matrix.ext }}"
```
Comment (Thegaram) [inline .github/workflows/releases.yaml]:
```suggestion
        binary_name="l2geth-${GOOS}-${GOARCH}${{ matrix.ext }}"
```

File: .github/workflows/releases.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      with:
+        draft: false
+        prerelease: false
+        # note you'll typically need to create a personal access token
+        # with permissions to create releases in the other repo
+        token: ${{ secrets.GITHUB_TOKEN }}
+        files: |
+          build/bin/geth-${{ matrix.goos }}-${{ matrix.goarch }}${{ matrix.ext }}
+          build/bin/geth-${{ matrix.goos }}-${{ matrix.goarch }}${{ matrix.ext }}.tar.gz
+          build/bin/geth-${{ matrix.goos }}-${{ matrix.goarch }}${{ matrix.ext }}.zip
```
Comment (Thegaram) [inline .github/workflows/releases.yaml]: Do we need all 3? I think 1 is probably enough.

File: .github/workflows/releases.yaml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
```
Comment (Thegaram) [inline .github/workflows/releases.yaml:65]: TODO: Consider adding a checksum (we can add this later)

File: .github/workflows/releases.yaml (hunk 3 above)
Comment (rrzhang139) [reply .github/workflows/releases.yaml]: I just followed what current assets are on releases, which is a zip and tar file. Which one should I leave?

Comment (rrzhang139) [reply .github/workflows/releases.yaml]: Wouldn't we want to package release binary so users can easily download it? Or is there no different from just posting the raw binary file?

**Pull request closed without merging**